six/modules/c++/cphd/include/cphd/CPHDXMLControl.h

```diff
@@ -71,7 +70,7 @@ class CPHDXMLControl : public six::XMLParser
      *  Calls toXML
      *  \return XML String
      */
-    std::string toXMLString(
+    virtual std::string toXMLString(
```

Comment:
Why are these virtual? Future-proofing?

Would it make sense to inherit from `six::XMLControl`?

Reply:
Yeah future-proofing was the intent.

So I tried inheriting from XMLControl, but XMLControl assumes the metadata is a six::Data object instead of the metadata object which I'm using.
Other than the validation function, which is static, I wouldn't be reusing/overriding anything in the XMLControl class.

Reply:
Aargh, that's annoying, but yeah makes sense.